python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py

```diff
@@ -38,6 +44,34 @@
 from lsst.pipe.base import AlgorithmError
 
 
+class PsfexNoStarsError(AlgorithmError):
+    """Raised if no stars are available for PSF determination."""
+
+    def __init__(self) -> None:
+        super().__init__("No psf candidates supplied.")
+
+
+class PsfexNoGoodStarsError(AlgorithmError):
+    """Raised if no "good" stars are available for PSF determination.
+
+    Parameters
+    ----------
+    num_available_stars : `int`
+        Number of available stars for PSF determination.
+    """
+
+    def __init__(self, num_available_stars) -> None:
+        self._num_available_stars = num_available_stars
+        super().__init__("No good psf candidates to pass to psfex.")
+
+    @property
+    def metadata(self) -> dict:
+        return {
+            "num_available_stars": self._num_available_stars,
+            "num_good_stars": 0,
```

Comment:
Again, `num_good_stars` isn't telling us anything that the exception itself already has.

And I think `num_available_stars` should maybe be `n_candidates`, or something like that, since that's where it's coming from (the size of the original candidate list).

```suggestion
            "n_candidates": self._n_candidates,
```

Reply:
I can remove `num_good_stars`, but regarding renaming `num_available_stars`, shouldn't we aim for consistent naming across our PSFEx exceptions? Using different names might confuse users.

I was trying to follow the pattern set by `PsfexTooFewGoodStarsError`:
```python
    @property
    def metadata(self) -> dict:
        return {
            "num_available_stars": self._num_available_stars,
            "num_good_stars": self._num_good_stars,
            "poly_ndim_initial": self._poly_ndim_initial,
            "poly_ndim_final": self._poly_ndim_final,
        }
```

Reply:
Ah, hmmm. If that's used in the other exception, then I guess we should keep it consistent.